refactor(sdk-coin-atom): use abstract-cosmos

hitansh-madan opened the pull request:
Ticket: WIN-1192

mmcshinsky-bitgo commented on modules/sdk-coin-atom/src/lib/index.ts:
```diff
@@ -1,12 +1,10 @@
+  CosmosTransactionBuilder as TransactionBuilder,
+} from '@bitgo/abstract-cosmos';
 export { KeyPair } from './keyPair';
-export { Transaction } from './transaction';
```
Removal of all these exports completely from here is a breaking change since /src/index.ts exposes these to end users.

hitansh-bitgo replied:
Added some of the exports back from abstract-cosmos & added a breaking change footer in the commit message for the ones still being removed completely

DinshawKothari commented on modules/sdk-coin-atom/src/lib/constants.ts:
```diff
@@ -1,12 +1,5 @@
 export const validDenoms = ['natom', 'uatom', 'matom', 'atom'];
 export const accountAddressRegex = /^(cosmos)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']{38})$/;
 export const validatorAddressRegex = /^(cosmosvaloper)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']{38})$/;
-export const contractAddressRegex = /^(cosmos)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']+)$/;
```
why removing contractAddressRegex ?

hitansh-bitgo replied:
Contract support was only added in abstract-cosmos & other chains but wasn't added for Atom it seems, so this was unused. I've updated the PR with the contract support changes needed for atom as well.

DinshawKothari commented on modules/statics/src/base.ts:
```diff
@@ -1438,7 +1438,7 @@ export enum KeyCurve {
  */
 export enum BaseUnit {
-  ATOM = 'uATOM',
+  ATOM = 'uatom',
```
Will this break anything in WP or we use lowercase there ?

hitansh-bitgo replied:
We were using lowercase everywhere except here

hitansh-madan committed:
refactor(sdk-coin-atom): use abstract-cosmos

Ticket: WIN-1192

BREAKING CHANGE: restructured exports
- Removed constants overlapping with @bitgo/abstract-cosmos
- Removed common Interfaces exported by @bitgo/abstract-cosmos

Code dependent on the previously exported Interfaces/Constants needs to
be updated to import these from @bitgo/abstract-cosmos instead
